### PR TITLE
Migrated test_volume_delete

### PR DESCRIPTION
**Pull request by nik-redhat**
Migrated TC [test_volume_delete](https://github.com/gluster/glusto-tests/blob/master/tests/functional/glusterd/test_volume_delete.py)

Fixes: #543 
Updates: #292 
Signed-off-by: nik-redhat <nladha@redhat.com>

### CODE REVIEW

**Comment by nik-redhat** [comment]
Test run:
```
Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_volume_delete
 
+-------------+-------------+--------------------+
| Volume Type | Test Result |  Time taken (sec)  |
+-------------+-------------+--------------------+
|     dist    |     PASS    | 31.37527847290039  |
|     rep     |     PASS    | 27.805692434310913 |
|   dist-rep  |     PASS    | 31.85454273223877  |
|     disp    |     PASS    | 25.076035022735596 |
|  dist-disp  |     PASS    | 32.75193929672241  |
+-------------+-------------+--------------------+



Framework runtime : 151.89226484298706
```

File: common/ops/gluster_ops/volume_ops.py [hunk 1]
```diff
@@ -249,8 +249,12 @@ def volume_delete(self, volname: str, node: str = None,
         cmd = f"gluster volume delete {volname} --mode=script --xml"
 
         ret = self.execute_abstract_op_node(cmd, node, excep)
-        self.es.add_data_to_cleands(self.es.get_brickdata(volname))
-        self.es.remove_volume_data(volname)
+
+        # Delete volume fro volds only if the command succeded
```
Comment (srijan-sivakumar) [inline common/ops/gluster_ops/volume_ops.py:253]: nit: from

File: tests/functional/glusterd/test_volume_delete.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        """
+        try:
+            # start glusterd on all servers
+            self.redant.start_glusterd(self.server_list)
+
+            if not self.redant.wait_for_glusterd_to_start(self.server_list):
+                raise Exception("Failed to start glusterd on all servers")
+
+            # Start volume if stopped
+            if not self.vol_started:
```
Comment (srijan-sivakumar) [inline tests/functional/glusterd/test_volume_delete.py:44]: We have the `cleanup_volume` inside the d_parent test which does the volume start if it hasn't. I think we can let that handle this situation ?

**Review by srijan-sivakumar — CHANGES_REQUESTED** [review]

File: tests/functional/glusterd/test_volume_delete.py (hunk 2 above)
Comment (nik-redhat) [reply tests/functional/glusterd/test_volume_delete.py:44]: Ack

File: common/ops/gluster_ops/volume_ops.py (hunk 1 above)
Comment (nik-redhat) [reply common/ops/gluster_ops/volume_ops.py:253]: :p

**Review by srijan-sivakumar — DISMISSED** [review]
LGTM

File: tests/functional/glusterd/test_volume_delete.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        if ret['msg']['opRet'] == 0:
+            raise Exception("Volume delete succeeded when one of the"
+                            " brick node is down")
+        if (re.search(r'Some of the peers are down', ret['msg']['opErrstr'])
+           is None):
+            raise Exception("Volume delete failed with unexpected error"
+                            " message")
+
+        redant.volume_start(self.vol_name, self.server_list[0])
+        self.vol_started = True
```
Comment (aujjwal-redhat) [inline tests/functional/glusterd/test_volume_delete.py:73]: nit: can we put this in tear Down. Just a suggestion.

Comment (nik-redhat) [reply tests/functional/glusterd/test_volume_delete.py:73]: This is the last step, so for the test to be successful we should be starting it at the end.
And, if anything goes wrong the terminate is already able to handle it.

Comment (aujjwal-redhat) [reply tests/functional/glusterd/test_volume_delete.py:73]: In the test case Ig they have put that part in the tear down
```js
def tearDown(self):
        # start the volume, it should succeed  
       ret, _, _ = volume_start(self.mnode, self.volname) 
       self.assertEqual(ret, 0, "Volume stop failed")<br class="Apple-interchange-newline">
```
Well, this is fine as well. Ig we can move ahead : )

Comment (nik-redhat) [reply tests/functional/glusterd/test_volume_delete.py:73]: Ohh :thinking: ..... I guess then we can remove this part as well, my bad.

Comment (nik-redhat) [reply tests/functional/glusterd/test_volume_delete.py]: Done

**Review by srijan-sivakumar — APPROVED** [review]

**Review by aujjwal-redhat — APPROVED** [review]
LGTM : )

**Pull request merged**